### PR TITLE
Added dimension computation

### PR DESCRIPTION
**Pull request by dmitriy-serdyuk**
It should be useful for lazy initialization.

### CODE REVIEW

File: blocks/bricks/conv.py [hunk 1]
```diff
@@ -148,13 +148,26 @@ class ConvolutionalLayer(Sequence, Initializable):
...
+                                         num_channels, conv_step, border_mode)
+        self.pooling = MaxPooling(pooling_size, pooling_step)
         super(ConvolutionalLayer, self).__init__(
-            application_methods=[convolution.apply, activation, pooling.apply],
+            application_methods=[self.convolution.apply, activation,
+                                 self.pooling.apply],
             **kwargs)
 
+    @property
+    def output_dim(self):
```
Comment (rizar) [inline blocks/bricks/conv.py]: In fact this is typically done by `get_dim` function in the rest of the library.

Comment (dmitriy-serdyuk) [inline blocks/bricks/conv.py]: I was confused by `MLP` class. Is it done differently there?

Comment (rizar) [inline blocks/bricks/conv.py]: In the MLP class `output_dim` is there to set dimension, but the main interface for asking them is `get_dim` (see `Brick` documentation). There is certain philosophy behind all this and I swear to explain it tomorrow! :)

File: blocks/bricks/conv.py [hunk 2]
```diff
@@ -147,14 +147,33 @@ class ConvolutionalLayer(Sequence, Initializable):
...
-            application_methods=[convolution.apply, activation, pooling.apply],
+            application_methods=[self.convolution.apply, activation,
+                                 self.pooling.apply],
             **kwargs)
 
+    def get_dim(self, name):
+        if name == 'input_':
+            return self.input_dim
+        if name == 'output':
+            if hasattr(self, '_output_dim'):
```
Comment (bartvm) [inline blocks/bricks/conv.py]: I think it's okay to repeat this computation, no need to store it in a private variable.

**Pull request merged**